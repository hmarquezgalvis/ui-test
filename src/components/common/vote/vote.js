import React from 'react';
import styled from 'styled-components';
import { ThumbDownIcon, ThumbUpIcon } from '../icons/thumbs';

export const VoteButton = ({ children, thumbDown = false, selected = false, value = null, onClick = () => {} }) => {
  const toggle = () => {
    onClick(value);
  };

  return (
    <ButtonContainer
      selected={selected}
      thumbDown={thumbDown}
      onClick={() => toggle(value)}
    >
      {thumbDown ? <Vote.ThumbDown /> : <Vote.ThumbUp />}
    </ButtonContainer>
  );
};
VoteButton.ThumbUp = ThumbUpIcon;
VoteButton.ThumbDown = ThumbDownIcon;

export const Vote = ({ children, thumbDown = false, value = null, onClick = () => {} }) => {
  return (
    <NormalContainer thumbDown={thumbDown}>
      {thumbDown ? <Vote.ThumbDown /> : <Vote.ThumbUp />}
    </NormalContainer>
  );
};
Vote.ThumbUp = ThumbUpIcon;
Vote.ThumbDown = ThumbDownIcon;

const ButtonContainer = styled.button`
  display: flex;
  width: 30px;
  height: 30px;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid ${props => props.selected ? '#ffffff' : 'transparent' };
  background: ${props => props.thumbDown ? "rgba(251, 189, 74, 1)" : "rgba(60, 187, 180, 0.8)" };
  cursor: pointer;
`;

const NormalContainer = styled.div`
  display: flex;
  width: 30px;
  height: 30px;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  background: ${props => props.thumbDown ? "rgba(251, 189, 74, 1)" : "rgba(60, 187, 180, 0.8)" };
  cursor: default;
`;