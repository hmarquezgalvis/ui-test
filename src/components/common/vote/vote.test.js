import { render } from '@testing-library/react';
import { Vote } from './vote';

describe('Tests for <Vote />', () => {

  describe('should render and save the snapshot correctly', () => {
    it('default state', () => {
      const { asFragment } = render(<Vote />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('marked state', () => {
      const { asFragment } = render(
        <Vote thumbDown={true} />
      );
      expect(asFragment()).toMatchSnapshot();
    });

  });

});