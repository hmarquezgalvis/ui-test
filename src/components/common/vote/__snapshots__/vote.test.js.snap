// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Vote /> should render and save the snapshot correctly default state 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi jJJlOR"
  >
    <img
      alt=""
      src="thumbs-up.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`Tests for <Vote /> should render and save the snapshot correctly marked state 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi jQxoBS"
  >
    <img
      alt=""
      src="thumbs-down.svg"
    />
  </div>
</DocumentFragment>
`;
